llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3463,7 +3463,7 @@ def : GCNPat <
              SRCMODS.NONE,
              (V_MOV_B64_PSEUDO (i64 0x3fefffffffffffff))),
          $x,
-         (V_CMP_CLASS_F64_e64 SRCMODS.NONE, $x, (i32 3 /*NaN*/))))
+         (V_CMP_CLASS_F64_e64 SRCMODS.NONE, $x, SRCMODS.NONE, (i32 3 /*NaN*/))))
```

Comment:
This shouldn't have gained a modifier operand

Reply:
Hi Matt. Since this patch change the instruction format from `f16, i16` to `f16, f16`, there is an addtional modifier operand for the floating point

Reply:
Why "f16, f16"? Can you avoid adding this, it is not useful here

Reply:
Hi Matt. The context of this change is that the `v_cmp_class_f16` has f16 in the src1 type. We change this to `f16` to enable inline of both int and floating points immediates. However it seems we could not find a way to suppress the additonal src modifier. also @Sisyph to help clarify if I misunderstood anything

Reply:
I think I misunderstood some part of the code when I took a closer look. Let me trim this patch and repost

Reply:
Modified this patch so that 32 and 64 bit instructions are not impacted